project-ideas/fulfillment-center-mgmt/readme.md

```diff
@@ -143,9 +143,100 @@ The TO is an OrderType like SalesOrder and PurchaseOrder.
 The API for to [createTransferOrder](createTransferOrder.md) builds on the [createOrder](../oms/createOrder.md)
 An inventory storage location may receive [InTransferShipment](createInTransferShipment.md) or ship [OutTransferShipment](createOutTransferShipment.md) for a transfer order. 
 
-
-
-
+### Scenarios
+The Transfer Orders will facilitate movement of inventory between locations, the scenarios being:
+1. TOs where Fulfillment location is managed by OMS and Receiving location is managed by third party e.g. Store to Warehouse
+2. TOs where Fulfillment location is managed by third party and Receiving location is managed by OMS e.g. Warehouse to Store
+3. TOs where both Fulfillment and Receiving locations are managed by OMS e.g. Store to Store
+
+### Design
+1. The Create Transfer Orders request to OMS will include an indicator like a **statusFlowId** on the basis of which OMS can decide whether the Transfer Order
+should be only fulfilled in OMS or only received in OMS or should be both fulfilled and received in OMS.
+2. This statusFlowId indicator will help in correctly transitioning the Transfer Order Item status and hence facilitate the transfer order 
+fulfillment and receiving as required in the system.
+3. The statusFlowId will be managed at OrderHeader level and new Order Item status and Status Flow Transitions will be added to manage the life cycle 
+of Transfer Order.
+
+#### Valid Status for Transfer Orders
+
+Two new Order Item Status will be introduced to manage the lifecycle of Transfer Orders which are PENDING_FULFILLMENT and PENDING_RECEIPT.
+
+1. Order Header - Transfer Order
+   1. Created
+   2. Approved
+   3. Completed
+   4. Cancelled
+   
+2. Order Item - Transfer Order Item
+   1. Created
+   2. Pending Fulfillment - newly introduced
+   3. Pending Receipt - newly introduced
+   4. Completed
+   5. Cancelled
+
+#### Transfer Order Life cycle
+
+1. TOs where Fulfillment location is managed by OMS and Receiving location is managed by third party e.g. Store to Warehouse
+   1. statusFlowId="TO_PendingFulfill"
+   2. Flow
+      1. TO Created in OMS
+         1. TO and TO Items in CREATED status
+      2. TO Approval
```

Comment:
As per discussion with @dixitdeepak Sir, his suggestion is to that approve service should not handle any status transitions to Pending Receive or Pending Fulfill, it should approve both order and items, and reserve inventory.

Since we do not want reservations to happen when TO is Pending Receive Only, we need to discuss how to handle this and status item updates after approval process.